### PR TITLE
[MAINTENANCE] Rename `profiler` folder and add readme with experimental note

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Rename `profiler` folder and add readme with experimental note
- Add docstring to Alice fixture and reorganize fixtures
- See Jira GE-134

### CODE REVIEW

File: great_expectations/rule_based_profiler/domain_builder/simple_semantic_type_domain_builder.py [hunk 1]
```diff
@@ -122,52 +122,40 @@ def infer_semantic_domain_type_from_table_column_type(
...
-                [
-                    type_name.upper()
-                    for type_name in ProfilerTypeMapping.RECORD_TYPE_NAMES
-                ]
-            )
+            {
+                type_name.upper()
+                for type_name in ProfilerTypeMapping.MISCELLANEOUS_TYPE_NAMES
+            }
+            | {type_name.upper() for type_name in ProfilerTypeMapping.RECORD_TYPE_NAMES}
```
Comment (alexsherstinsky) [inline great_expectations/rule_based_profiler/domain_builder/simple_semantic_type_domain_builder.py:158]: Nice!

File: tests/rule_based_profiler/alice_user_workflow_fixture.py [hunk 2]
```diff
@@ -11,6 +11,25 @@
 
 @pytest.fixture
 def alice_columnar_table_single_batch():
+    """
```
Comment (alexsherstinsky) [inline tests/rule_based_profiler/alice_user_workflow_fixture.py:14]: Beautiful!

File: tests/rule_based_profiler/test_profiler_user_workflows.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
+import os
```
Comment (alexsherstinsky) [inline tests/rule_based_profiler/test_profiler_user_workflows.py:1]: I like this refactor!

**Review by alexsherstinsky — APPROVED** [review]
LGreatTM :)

File: great_expectations/rule_based_profiler/domain_builder/simple_semantic_type_domain_builder.py (hunk 1 above)
Comment (anthonyburdi) [reply great_expectations/rule_based_profiler/domain_builder/simple_semantic_type_domain_builder.py:158]: You wrote this part so it sure is Nice! 😄

**Pull request merged**